lib/ansible/modules/identity/cyberark/cyberark_credential.py

```diff
@@ -0,0 +1,259 @@
+#!/usr/bin/python
+# Copyright: (c) 2017, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+module: cyberark_credential
+short_description: Module for retrieval of CyberArk vaulted credential using PAS Web Services SDK through the Central Credential Provider
+author:
+ - Edward Nunez (@enunez-cyberark)
+ - Cyberark Bizdev (@cyberark-bizdev)
+ - erasmix (@erasmix)
+ - James Stutes (@jimmyjamcabd)
+version_added: 2.8
+description:
+    - Creates a URI for retrieving a credential from the Cyberark Vault through the Privileged
+      Account Security Web Services SDK by requesting access to a specific object through an Application ID
+      It returns an Ansible fact called I(cyberarkcredential) as a JSON message with object information
+      that can be used by other modules. Every module can use this fact as C(cyberarkcredential) parameter.
+options:
+    api_base_url:
+        description:
+            - A string containing the base URL of the server hosting the Central Credential Provider
+    validate_certs:
+        type: bool
+        default: 'true'
+        description:
+            - If C(false), SSL certificate chain will not be validated.  This should only
+              set to C(true) if you have a root CA certificate installed on each node.
+    app_id:
+        description:
+            - A string containing the Application ID authorized for retrieving the credential.
+    query:
+        description:
+            - A string containing details of the object being queried
+              parameters
+                Safe=<safe name>
+                Folder=<folder name within safe>
+                Object=<object name>
+                UserName=<username of object>
+                Address=<address listed for object>
+                Database=<optional file category for database objects>
+                PolicyID=<platform id managing object>.
+        required: True
+    client_cert:
+        required: False
+        description:
+            - A string containing the file location and name of the client certificate used for authentication.
+    client_key:
+        required: False
+        description:
+            - A string containing the file location and name of the private key of the client certificate used for authentication.
+    reason:
+        required: False
+        description:
+            - Reason for requesting credential if required by policy.
+    state:
```

Comment:
Will there be other `states` in the future?

Reply:
For this particular module the state will always be present, there is no workflow for absent.

Reply:
Why have a state option at all then if there is only one value allowed?

Reply:
I had originally left it off and I want to say that during lint testing it was added back.  I think it was conflicting with either documentation or another setting.  I would prefer to leave it off.